pkg/platform/runtime/store/store.go

```diff
@@ -70,22 +72,41 @@ func (s *Store) HasMarker() bool {
 	return false
 }
 
-// MatchesCommit checks if stored runtime is complete and can be loaded
-func (s *Store) MatchesCommit(commitID strfmt.UUID) bool {
+// MarkerIsValid checks if stored runtime is complete and can be loaded
+func (s *Store) MarkerIsValid(commitID strfmt.UUID) bool {
 	marker := s.markerFile()
 	if !fileutils.FileExists(marker) {
 		logging.Debug("Marker does not exist: %s", marker)
 		return false
 	}
 
-	contents, err := fileutils.ReadFile(marker)
+	f, err := os.OpenFile(marker, os.O_RDONLY, 0)
```

Comment:
I don't think we should be relying on the lower level implementations just so we can go line by line. Going line by line with just a string/bytearray as input is pretty simple, don't need to rely on IO libraries for that.